tests/testthat/test-local.R

```diff
@@ -1,10 +1,28 @@
-test_that("expected options are set by default", {
+test_that("local context is 'as promised' inside test_that()", {
+  # high-level expectations or expectations that preceded testthat 3e
   expect_true(is_testing())
   expect_equal(testing_package(), "testthat")
   expect_false(is_interactive())
   expect_equal(Sys.getenv("R_TESTS"), "")
+
+  # set in local_test_context()
+  expect_equal(Sys.getenv("TESTTHAT"), "true")
+
+  # testthat 3e, set in local_reproducible_output()
```

Comment:
These are documented as being set by `local_test_context()`, but are actually set by `local_reproducible_output()`, which seems a bit confusing.

  * `cli.unicode`
  * `crayon.enabled`
  * `width`

 But `local_test_context()` calls `local_reproducible_output()` if using 3e, which points out that maybe I don't fully understand the ways in which all of these settings are actually changed inside `test_that()`.